pydantic/_internal/_validate_call.py

```diff
@@ -120,6 +120,11 @@ def __get__(self, obj: Any, objtype: type[Any] | None = None) -> ValidateCallWra
         if obj is None:
             try:
                 # Handle the case where a method is accessed as a class attribute
+                # Its possible this wrapper is dynamically applied to a class attribute not allowing
+                # name to be populated by __set_name__. In this case, we'll manually acquire the name
+                # from the function reference.
+                if self._name is None:
+                    self._name = self.raw_function.__name__
```

Comment:
Looks great. Let's move this outside of the `try` block, but keep it within the scope of the `if obj is None` check.

Reply:
done !